File: c/src/armstrong_numbers.c
#include <math.h>
#include "armstrong_numbers.h"
bool is_armstrong_number(int num) {
    int len = 1 + log10(num);

    int total = 0;
    int working = num;
    while (working > 0) {
        total += pow(working%10, len);
        working /= 10;
    }

    return num == total;
}

File: java/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(final int numberToCheck) {
        if (numberToCheck < 0) {
            throw new IllegalArgumentException("You must supply a non-negative integer");
        }
        int total = 0;
        if (numberToCheck > 0) {
            final int power = 1 + (int)Math.log10(numberToCheck);
            int working = numberToCheck;
            while (working > 0) {
                total += Math.pow(working % 10, power);
                working = working / 10; 
            }
        }
        return total == numberToCheck;

    }

}
